api/src/groups/test_groups.py

```diff
@@ -5,993 +5,816 @@
     groups = auth_client.sqla.query(Group).all()
-
-    # WHEN group is changed to active
+    # THEN list one group section of group
     for group in groups:
-        resp = auth_client.put(url_for('groups.activate_group', group_id=group.id),
-                               json={'active': True})
-        # THEN assert group is active
+        resp = auth_client.get(url_for(
+            'groups.read_one_group', group_id=group.id))
         assert resp.status_code == 200
-        assert resp.json['active'] == True
-
-    resp = auth_client.put(url_for('groups.activate_group', group_id=500), json={'active': True})
-    # THEN assert group is not found
-    assert resp.status_code == 404
+        assert resp.json['name'] == group.name
+        assert resp.json['description'] == group.description
+        assert resp.json['active'] == group.active
+        assert resp.json['groupTypeId'] == group.group_type_id
 
 
 @pytest.mark.smoke
-def test_deactivate_group(auth_client):
-    # GIVEN group to deactivate
+def test_update_group(auth_client):
+    # Test with invalid group
+    # GIVEN empty database
+    # WHEN databse queried
+    resp = auth_client.patch(
+        url_for('groups.update_group', group_id=555))
+    # THEN assert error code
+    assert resp.status_code == 422
+    # Test with populated database
+    # GIVEN an existing (active or inactive) group
+    create_multiple_group_types(auth_client.sqla, 1)
     count = random.randint(3, 11)
-    generate_managers(auth_client)
     create_multiple_groups(auth_client.sqla, count)
-
     groups = auth_client.sqla.query(Group).all()
-
-    # WHEN group is changed to inactive
+    # WHEN group needs to update existing information
     for group in groups:
-        resp = auth_client.put(url_for('groups.deactivate_group', group_id=group.id),
-                               json={'active': False})
-        # THEN assert group is inactive
+        resp = auth_client.patch(url_for('groups.update_group', group_id=group.id),
+                                 json={'name': 'asd', 'description': 'test_descr', 'active': False})
+        # THEN assert changes to group  reflect update
         assert resp.status_code == 200
+        assert resp.json['name'] == 'asd'
+        assert resp.json['description'] == 'test_descr'
         assert resp.json['active'] == False
-
-    resp = auth_client.put(url_for('groups.deactivate_group', group_id=500), json={'active': False})
-    # THEN assert group is not found
-    assert resp.status_code == 404
-
-
-# ---- Meeting
-
-
+# 
+# @pytest.mark.smoke
+# def test_invalid_update_group(auth_client):
+#     pass
+#
+@pytest.mark.smoke
+def test_subset_group(auth_client):
+    # Test with invalid subset with "offset"
+    # GIVEN empty database
+    # WHEN databse queried
+    create_multiple_group_types(auth_client.sqla, 1)
+    count = random.randint(3, 11)
+    create_multiple_groups(auth_client.sqla, count)
+    # Test with populated database
```

Comment:
Refactor into own test function. 

Take a look for other places where multiple tests have been lumped into a single function and separate them.